Source/WebKitLegacy/mac/WebView/WebView.mm

```diff
@@ -2267,10 +2267,8 @@ - (void)_dispatchTileDidDraw:(CALayer*)tile
         return;
     }
 
-ALLOW_DEPRECATED_DECLARATIONS_BEGIN
-    if (!OSAtomicCompareAndSwap32(0, 1, &_private->didDrawTiles))
+    if (!WTF::atomicCompareExchangeStrong(&_private->didDrawTiles, NO, YES, std::memory_order_relaxed, std::memory_order_relaxed))
```

Comment:
...although I'm not confident I understand memory order well enough, and although I also see you preserved the preexisting behavior here of not using any memory fence, I suspect the existing code was wrong and you actually _do_ require a memory barrier here, right? I think you need `std::memory_order_acq_rel` in the success case and `std::memory_order_acquire` in the failure case.

Having any mention of memory order makes it more difficult to reason about the correctness of the code, so it might be best to optimize for readability instead of performance and just use the default sequential consistent memory order, even if it's stricter than what is truly needed.